pkg/scheduler/core/generic_scheduler.go

```diff
@@ -673,12 +663,15 @@ func podFitsOnNode(
 			if predicate, exist := predicateFuncs[predicateKey]; exist {
 				fit, reasons, err = predicate(pod, metaToUse, nodeInfoToUse)
 				if err != nil {
-					return false, []predicates.PredicateFailureReason{}, err
+					return false, []*framework.Status{}, err
```

Comment:
Here, we should return `framework.Error` for the Status.Code. We can possibly get rid of `err`.

Reply:
> Here, we should return `framework.Error` for the Status.Code. We can 
possibly get rid of `err`.

Is it necessary to do it here? I found that a little tricky to differentiate the case between `err != nil` and `!fits` after mixing them since they we use a different approach to deal with them.

```go
			fits, failedPredicates, err := podFitsOnNode(
				pod,
				meta,
				g.nodeInfoSnapshot.NodeInfoMap[nodeName],
				g.predicates,
				g.schedulingQueue,
				g.alwaysCheckAllPredicates,
			)
			if err != nil {
				predicateResultLock.Lock()
				errs[err.Error()]++
				predicateResultLock.Unlock()
				return
			}
			if !fits {
				predicateResultLock.Lock()
				failedFilterMap[nodeName] = failedPredicates
				predicateResultLock.Unlock()
				return
			}
```